### PR TITLE
Added Api for the tournaments

### PR DESCRIPTION
**Pull request by AayushSaini101**
Added Api for the tournaments

### CODE REVIEW

File: controller/tournamentController.js [hunk 1]
```diff
@@ -1 +1,56 @@
...
+            } else {
+                Logger.info('Tournament created successfully');
+                res.send(tournament);
+            }
+        });
+
+
+    }
+// Function to view the tournament details
+function viewTournamentDetails(req, res) {
```
Comment (Aditya-Thakur) [inline controller/tournamentController.js:33]: change it to viewAllTournament

File: controller/tournamentController.js [hunk 2]
```diff
@@ -1 +1,56 @@
...
+        } else {
+            Logger.info('Tournament details fetched successfully');
+            res.send(tournament);
+        }
+    });
+}
+
+//Function to join the tournament using the tournament ID
+function joinTournament(req, res) {
+    tournament.findOne({ tournamentId: req.body.tournamentId }, function(err, tournament) {
```
Comment (Aditya-Thakur) [inline controller/tournamentController.js:47]: This function needs to be like this: 
function should get userId & tournamentId. 
It should update both user & tournament collection

File: controller/tournamentController.js [hunk 3]
```diff
@@ -1 +1,56 @@
-// ToBe Implemented
\ No newline at end of file
+// ToBe Implemented
+const mongoose = require('mongoose');
+const Logger = require('nodemon/lib/utils/log');
+const user = mongoose.model('User');
```
Comment (Aditya-Thakur) [inline controller/tournamentController.js:4]: require tournament here not user

File: controller/tournamentController.js [hunk 4]
```diff
@@ -1 +1,56 @@
...
+
+function createNewTournament(req, res) {
+    console.log('need to register ', req.body.email);
```
Comment (Aditya-Thakur) [inline controller/tournamentController.js:7]: remove this log since you are using logger

**Review by Aditya-Thakur — CHANGES_REQUESTED** [review]
Please go through the comments and update it.

**Comment by Aditya-Thakur** [comment]
Raise another PR after changes.

**Comment by AayushSaini101** [comment]
Okay Sir

**Pull request merged**